src/main/java/cmd/CmdParser.java

```diff
@@ -110,6 +110,14 @@ private Configuration parseParameters(CommandLine cmd) {
             }
         }
 
+        if (cmd.hasOption("g")) {
+            if (config.type == ComparatorType.XY_STR) {
+                config.path = cmd.getOptionValue("g");
+            } else {
+                throw new CommandLineException("Incorrect user input! Type [-h] or [--help] for help!");
```

Comment:
It's better to extract string to the constant field: it is the same in a few places.